activity_browser/app/ui/tables/inventory.py

```diff
@@ -151,45 +176,59 @@ def __init__(self, parent=None):
-        self.add_activity_action = QtWidgets.QAction(
+        self.open_activity_action = QtWidgets.QAction(
+            QtGui.QIcon(icons.left), "Open activity", None)
+        self.new_activity_action = QtWidgets.QAction(
             QtGui.QIcon(icons.add), "Add new activity", None
         )
-        self.copy_activity_action = QtWidgets.QAction(
-            QtGui.QIcon(icons.copy), "Copy activity", None
+        self.duplicate_activity_action = QtWidgets.QAction(
+            QtGui.QIcon(icons.copy), "Duplicate activity", None
         )
         self.delete_activity_action = QtWidgets.QAction(
             QtGui.QIcon(icons.delete), "Delete activity", None
         )
-        self.open_left_tab_action = QtWidgets.QAction(
-            QtGui.QIcon(icons.left), "Open in new tab", None
-        )
-        self.copy_to_db_action = QtWidgets.QAction(
-            QtGui.QIcon(icons.add_db), 'Copy to database', None
+        self.duplicate_activity_to_db_action = QtWidgets.QAction(
+            QtGui.QIcon(icons.add_db), 'Duplicate to other database', None
         )
-        self.addAction(self.add_activity_action)
-        self.addAction(self.copy_activity_action)
+        # context menu items are disabled if db is read-only
+
+        self.update_activity_table_context(self.database_name, db_read_only=self.db_read_only)
+
+        self.addAction(self.open_activity_action)
+        self.addAction(self.new_activity_action)
+        self.addAction(self.duplicate_activity_action)
         self.addAction(self.delete_activity_action)
-        self.addAction(self.open_left_tab_action)
-        self.addAction(self.copy_to_db_action)
-        self.add_activity_action.triggered.connect(
+        self.addAction(self.duplicate_activity_to_db_action)
+
+        self.open_activity_action.triggered.connect(
+            lambda x: signals.open_activity_tab.emit("activities", self.currentItem().key)
+        )
+        self.new_activity_action.triggered.connect(
             lambda: signals.new_activity.emit(self.database.name)
         )
-        self.copy_activity_action.triggered.connect(
-            lambda x: signals.copy_activity.emit(self.currentItem().key)
+        self.duplicate_activity_action.triggered.connect(
+            lambda x: signals.duplicate_activity.emit(self.currentItem().key)
         )
         self.delete_activity_action.triggered.connect(
             lambda x: signals.delete_activity.emit(self.currentItem().key)
         )
-        self.open_left_tab_action.triggered.connect(
-            lambda x: signals.open_activity_tab.emit("activities", self.currentItem().key)
-        )
-        self.copy_to_db_action.triggered.connect(
-            lambda: signals.copy_to_db.emit(self.currentItem().key)
+        self.duplicate_activity_to_db_action.triggered.connect(
+            lambda: signals.show_duplicate_to_db_interface.emit(self.currentItem().key)
         )
 
+    def update_activity_table_context(self, db, db_read_only):
```

Comment:
I like this a lot! Very neat how the options are only available for editable databases